src/crawlee/storages/_storage_instance_manager.py

```diff
@@ -48,100 +66,113 @@ async def open_storage_instance(
         id: str | None,
         name: str | None,
         alias: str | None,
-        configuration: Configuration,
+        storage_client_type: type[StorageClient],
         client_opener: ClientOpener,
+        additional_cache_key: Hashable = '',
     ) -> T:
         """Open a storage instance with caching support.
 
         Args:
             cls: The storage class to instantiate.
             id: Storage ID.
-            name: Storage name (global scope, persists across runs).
+            name: Storage name.
             alias: Storage alias (run scope, creates unnamed storage).
-            configuration: Configuration object.
-            client_opener: Function to create the storage client.
+            storage_client_type: Type of storage client to use.
+            client_opener: Coroutine to open the storage client when storage instance not found in cache.
+            additional_cache_key: Additional optional key to differentiate cache entries.
 
         Returns:
             The storage instance.
 
         Raises:
             ValueError: If multiple parameters out of `id`, `name`, and `alias` are specified.
         """
-        # Validate input parameters.
-        specified_params = sum(1 for param in [id, name, alias] if param is not None)
-        if specified_params > 1:
-            raise ValueError('Only one of "id", "name", or "alias" can be specified, not multiple.')
-
-        # Auto-set alias='default' when no parameters are specified.
-        # Default unnamed storage is equal to alias=default unnamed storage.
-        if specified_params == 0:
-            alias = self._DEFAULT_STORAGE
-            specified_params = 1
-
-        # Check cache
-        if id is not None:
-            type_cache_by_id = self._cache_by_id.get(cls, {})
-            if id in type_cache_by_id:
-                cached_instance = type_cache_by_id[id]
+        try:
+            if name == self._DEFAULT_STORAGE_ALIAS:
+                raise ValueError(
+                    f'Storage name cannot be "{self._DEFAULT_STORAGE_ALIAS}" as it is reserved for default alias.'
+                )
+
+            # Validate input parameters.
+            specified_params = sum(1 for param in [id, name, alias] if param is not None)
+            if specified_params > 1:
+                raise ValueError('Only one of "id", "name", or "alias" can be specified, not multiple.')
+
+            # Auto-set alias='default' when no parameters are specified.
+            # Default unnamed storage is equal to alias=default unnamed storage.
+            if specified_params == 0:
+                alias = self._DEFAULT_STORAGE_ALIAS
```

Comment:
We should probably normalize this in an opposite direction - when alias=default, let's use alias=None. Because of the implementation in the SDK - to not to create a record with the default mapping in the default KVS, because this is resolved via the DEFAULT_... env vars. Could you do it here please?

Reply:
Okay